addons/account/models/partner.py

```diff
@@ -167,11 +167,17 @@ def get_fiscal_position(self, partner_id, delivery_id=None):
         # This can be easily overridden to apply more complex fiscal rules
         PartnerObj = self.env['res.partner']
         partner = PartnerObj.browse(partner_id)
+        delivery = PartnerObj.browse(delivery_id)
 
-        # if no delivery use invoicing
-        if delivery_id:
-            delivery = PartnerObj.browse(delivery_id)
-        else:
+        company = self.env.company
+        eu_country_codes = self.env.ref('base.europe').country_ids.mapped('code')
+        has_vat = bool(company.vat) and bool(partner.vat)
+        if has_vat:
+            intra_eu = company.vat[:2] in eu_country_codes and partner.vat[:2] in eu_country_codes
+            vat_exclusion = company.vat[:2] == partner.vat[:2]
+
+        # If company and delivery have the same vat prefix, use invoicing
```

Comment:
So `delivery = partner` is never done outside of EU (outside of the case where `delivery_id` is not provided)?

Again, same question as before, why does EU rule the world?

Reply:
Because I think the use cases we are trying to cover here are related to EU specific fiscal rules ? :man_shrugging: